File: PermMissingElem/ruby/PermMissingElem.rb
class PermMissingElem 
    def solution(a)
        
         length = a.length + 1
         sum1 = 0
         sum2 = 0
                
         a.each { |x| sum1 += x }
        
         sum2 = (((length+1)*length)/2) - sum1
         sum2

    end
end

File: PermMissingElem/php/PermMissingElem.php
<?php

namespace Codility;

class PermMissingElem {
    function solution($A) {
        
         $length = count($A)+1;
         $sum1 = 0;
         $sum2 = 0;
         
         foreach ($A as $e) { $sum1 += $e; }       
         $sum2 = ((($length+1)*$length)/2);

         return intval($sum2 - $sum1);

    }
}